pkg/alertmanager/state_persister.go

```diff
@@ -0,0 +1,100 @@
+package alertmanager
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/alertmanager/cluster/clusterpb"
+
+	"github.com/cortexproject/cortex/pkg/alertmanager/alertspb"
+	"github.com/cortexproject/cortex/pkg/alertmanager/alertstore"
+	"github.com/cortexproject/cortex/pkg/util/services"
+)
+
+const (
+	defaultPersistInterval = 15 * time.Minute
```

Comment:
I would make this configurable (via CLI flag / YAML config). It's something a Cortex cluster operator may need to tune.